src/webchat-ui/components/WebchatUI.tsx

```diff
@@ -314,6 +314,47 @@ export class WebchatUI extends React.PureComponent<React.HTMLProps<HTMLDivElemen
             />
         );
     }
+    
+    handleKeydown = (event) => {
+        const closeButtonInHeader = document.getElementById("webchatHeaderCloseButton");
```

Comment:
Relying on document.getElementById seems kind of "hacky" as we might accidently collide with other elements on the page.

Would it be possible to use `refs` here? Then we would explicitly assign dom element instances within the webchat.

Reply:
Replaced ids with refs wherever possible.